pkg/agent/openflow/network_policy.go

```diff
@@ -863,57 +865,50 @@ func generateServiceConjMatches(ruleTableID uint8, service v1beta2.Service, prio
 	return matches
 }
 
-func getServiceMatchPairs(service v1beta2.Service, ipProtocols []binding.Protocol, matchSrc bool) [][]matchPair {
+func getServiceMatchPairs(service v1beta2.Service, ipProtocols []binding.Protocol) [][]matchPair {
 	var conjMatchesMatchPairs [][]matchPair
-	ovsBitRanges := serviceToBitRanges(service)
-	addL4MatchPairs := func(matchKey *types.MatchKey) {
+	ovsBitRanges := portsToBitRanges(service.Port, service.EndPort)
+	var srcOVSBitRanges []types.BitRange
+	if service.SrcPort != nil {
+		srcPortTyped := intstr.FromInt(int(*service.SrcPort))
+		srcOVSBitRanges = portsToBitRanges(&srcPortTyped, service.SrcEndPort)
+	}
+	addL4MatchPairs := func(matchKey, srcMatchKey *types.MatchKey) {
 		for _, ovsBitRange := range ovsBitRanges {
-			conjMatchesMatchPairs = append(conjMatchesMatchPairs, []matchPair{{matchKey: matchKey, matchValue: ovsBitRange}})
+			matchPairs := []matchPair{{matchKey: matchKey, matchValue: ovsBitRange}}
+			if srcOVSBitRanges != nil {
+				for _, srcRange := range srcOVSBitRanges {
+					matchPairs = append(matchPairs, matchPair{matchKey: srcMatchKey, matchValue: srcRange})
+					conjMatchesMatchPairs = append(conjMatchesMatchPairs, matchPairs)
+				}
+			} else {
```

Comment:
Can `port`+`endPort` and `srcPort`+`srcEndPort` be used together?

Reply:
They can. If you look closely, the `if` clause deals with the situation where both port/endPort and srcPort/srcEndPort are specified, while the `else` clause deals with only port/endPort

Reply:
Thanks for pointing it out.
Based on the definition of func `portsToBitRanges`, `srcOVSBitRanges` won't be nil I think you could remove this if/else.

Reply:
Ah, my bad. Thanks for pointing it out.

Reply:
Yes they can. We have E2E testcase covering that